File: src/main/java/io/github/bloodnighttw/whateverBot/codeWrapper/LanguageDetector.java
package io.github.bloodnighttw.whateverBot.codeWrapper;

import java.util.Arrays;

public class LanguageDetector {

	private final static String JAVA = "public static void main(String[]";
	private final static String[] CPP = {"#incluide <iostream>", "#include <bits/stdc++.h>"};
	private final static String C = "#include <stdio.h>";

	public static Language detectCode(String code) {
		if (code.contains(JAVA))
			return Language.Java;
		if (Arrays.stream(CPP).anyMatch(code::contains))
			return Language.CPP;
		if (code.contains(C))
			return Language.C;
		return Language.Other;
	}

}

File: src/main/java/io/github/bloodnighttw/whateverBot/codeWrapper/LanguageDetector.kt
package io.github.bloodnighttw.whateverBot.codeWrapper

private const val JAVA = "public static void main(String[]"
private const val CPP1 = "#incluide <iostream>"
private const val CPP2 = "#include <bits/stdc++.h>"
private const val C = "#include <stdio.h>"

fun detectCode(code: String): Language {
        if (JAVA in code) return Language.Java
        if (CPP1 in code || CPP2 in code) return Language.CPP
        return if (C in code) Language.C
        else Language.Other
}


